e2e-tests/playwright/docs/accessibility/interaction_testing.md

```diff
@@ -0,0 +1,405 @@
+    const expandButton = page.locator('[aria-expanded]');
+    const expandableContent = page.locator('#expandable-content');
+
+    // Test initial collapsed state
+    await expect(expandButton).toHaveAttribute('aria-expanded', 'false');
+    await expect(expandableContent).toBeHidden();
+
+    // Expand with keyboard
+    await expandButton.focus();
+    await page.keyboard.press('Enter');
+
+    // Verify expanded state
+    await expect(expandButton).toHaveAttribute('aria-expanded', 'true');
+    await expect(expandableContent).toBeVisible();
+
+    // Verify focus moves to content
+    await expect(expandableContent).toBeFocused();
+
+    // Collapse with keyboard
+    await expandButton.focus();
+    await page.keyboard.press('Enter');
+
+    // Verify collapsed state
+    await expect(expandButton).toHaveAttribute('aria-expanded', 'false');
+    await expect(expandableContent).toBeHidden();
+});
+```
+
+### **4. Form Validation Accessibility**
+
+```typescript
+test('form validation accessibility', async ({page}) => {
+    const form = page.locator('form');
+    const requiredField = form.locator('[required]');
+    const submitButton = form.locator('[type="submit"]');
+
+    // Submit empty form to trigger validation
+    await submitButton.focus();
+    await page.keyboard.press('Enter');
+
+    // Verify error state accessibility
+    await expect(requiredField).toHaveAttribute('aria-invalid', 'true');
+    await expect(page.locator('[role="alert"]')).toBeVisible();
+
+    // Verify focus moves to invalid field
+    await expect(requiredField).toBeFocused();
+
+    // Test error message association
+    const errorId = await page.locator('[role="alert"]').getAttribute('id');
+    await expect(requiredField).toHaveAttribute('aria-describedby', errorId);
+
+    // Fix validation error
+    await requiredField.fill('valid input');
+    await submitButton.focus();
+    await page.keyboard.press('Enter');
+
+    // Verify error cleared
+    await expect(requiredField).toHaveAttribute('aria-invalid', 'false');
+    await expect(page.locator('[role="alert"]')).toBeHidden();
+});
+```
+
+### **5. Live Region Announcements**
+
+```typescript
+test('dynamic content announcements', async ({page}) => {
+    const liveRegion = page.locator('[aria-live]');
+    const triggerButton = page.locator('[data-testid="load-content"]');
+
+    // Verify live region setup
+    await expect(liveRegion).toHaveAttribute('aria-live', 'polite');
+
+    // Trigger content update
+    await triggerButton.click();
+
+    // Wait for content to appear in live region
+    await expect(liveRegion).toContainText('Loading complete');
+
+    // Test urgent announcements
+    const urgentRegion = page.locator('[aria-live="assertive"]');
+    await page.click('[data-testid="urgent-action"]');
+    await expect(urgentRegion).toContainText('Critical update');
+});
+```
+
+## Advanced Testing Techniques
+
+### **1. Focus Visibility Testing**
```

Comment:
Could you remove this example? I don't think we should ever be writing tests that test this sort of thing because it's just verifying that the browser is behaving correctly